src/main/java/com/google/devtools/build/lib/exec/StandaloneTestStrategy.java

```diff
@@ -240,20 +241,26 @@ private StandaloneFailedAttemptResult processTestAttempt(
       testOutputs = renameOutputs(actionExecutionContext, action, testOutputs, attemptId);
     }
 
+    TestResultData.Builder dataBuilder = result.testResultDataBuilder();
+    // Start off incomplete, only a discovered test log permits a pass
+    BlazeTestStatus status = BlazeTestStatus.INCOMPLETE;
     // Recover the test log path, which may have been renamed, and add it to the data builder.
     Path renamedTestLog = null;
     for (Pair<String, Path> pair : testOutputs) {
       if (TestFileNameConstants.TEST_LOG.equals(pair.getFirst())) {
+        Preconditions.checkState(renamedTestLog == null, "multiple test_log matches");
         renamedTestLog = pair.getSecond();
+        status = dataBuilder.getStatus();
       }
     }
 
-    TestResultData.Builder dataBuilder = result.testResultDataBuilder();
-    if (dataBuilder.getStatus() == BlazeTestStatus.PASSED) {
+    if (status == BlazeTestStatus.PASSED) {
```

Comment:
I would do this slightly differently. I think it might be nicer to check explicitly for `renamedTestLog == null` here as the first case, and then override the status in dataBuilder.

However, I'm also not sure how this can happen in the first place - why doesn't the code realize that the action got cancelled?

Reply:
That's fine on the rework, I considered that but thought adding that check seemed a bit off - felt more consistent to have a status that we observed associated with the found TEST_LOG

I'm curious about that as well, but my guess is that some cancellation of a test strategy internal or post process is what triggered this after test cancellation detection took place. See the associated issue for proof of the capability to hit this itself.